File: backend/src/main/java/com/programmers/pcquotation/global/Properties/OAuth2ProviderProperties.java
package com.programmers.pcquotation.global.Properties;

import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "spring.security.oauth2.client.provider")
public record OAuth2ProviderProperties(
	ProviderConfig kakao,
	ProviderConfig naver
) {

	public record ProviderConfig(
		String authorizationUri,
		String tokenUri,
		String userInfoUri,
		String userNameAttribute
	) {
	}
}

File: backend/src/main/java/com/programmers/pcquotation/global/properties/OAuth2ProviderProperties.kt
package com.programmers.pcquotation.global.properties

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "spring.security.oauth2.client.provider")
data class OAuth2ProviderProperties(
    val kakao: ProviderConfig,
    val naver: ProviderConfig
) {
    data class ProviderConfig(
        val authorizationUri: String,
        val tokenUri: String,
        val userInfoUri: String,
        val userNameAttribute: String
    )
}